src/Castle.Windsor/MicroKernel/Registration/ServiceDescriptor.cs

```diff
@@ -229,14 +230,14 @@ private IEnumerable<Type> GetTopLevelInterfaces(Type type)
 		/// <returns></returns>
 		private static Type WorkaroundCLRBug(Type serviceType)
 		{
-			if (!serviceType.IsInterface)
+			if (!serviceType.GetTypeInfo().IsInterface)
 			{
 				return serviceType;
 			}
 			// This is a workaround for a CLR bug in
 			// which GetInterfaces() returns interfaces
 			// with no implementations.
-			if (serviceType.IsGenericType && serviceType.ReflectedType == null)
+			if (serviceType.GetTypeInfo().IsGenericType)
```

Comment:
Should we wrap ReflectedType in a #if? It is in net45, and also added in netstandard2.0.